pkg/koordlet/metricsadvisor/performance_collector_linux.go

```diff
@@ -134,6 +144,117 @@ func (c *performanceCollector) profilePerfOnSingleContainer(containerStatus *cor
+			FullCPUAvg10: psiMap["FullCPUAvg10"],
+			FullMemAvg10: psiMap["FullMemAvg10"],
+			FullIOAvg10:  psiMap["FullIOAvg10"],
+		},
+	}
+	err = c.metricCache.InsertContainerInterferenceMetrics(collectTime, containerPsiMetric)
+	if err != nil {
+		klog.Errorf("insert container psi metrics failed, err %v", err)
+	}
+	metrics.RecordContainerPSI(containerStatus.ContainerID, podUid, "some", "cpu", psiMap["SomeCPUAvg10"])
+	metrics.RecordContainerPSI(containerStatus.ContainerID, podUid, "some", "mem", psiMap["SomeMemAvg10"])
+	metrics.RecordContainerPSI(containerStatus.ContainerID, podUid, "some", "io", psiMap["SomeIOAvg10"])
+	metrics.RecordContainerPSI(containerStatus.ContainerID, podUid, "full", "cpu", psiMap["FullCPUAvg10"])
+	metrics.RecordContainerPSI(containerStatus.ContainerID, podUid, "full", "mem", psiMap["FullMemAvg10"])
+	metrics.RecordContainerPSI(containerStatus.ContainerID, podUid, "full", "io", psiMap["FullIOAvg10"])
+}
+
+func (c *performanceCollector) collectPodMetrics() {
+	klog.V(6).Infof("start collectPodMetrics")
+	timeWindow := time.Now()
+	podMetas := c.statesInformer.GetAllPods()
+	var wg sync.WaitGroup
+	wg.Add(len(podMetas))
+	for _, meta := range podMetas {
+		pod := meta.Pod
+		podCgroupDir := meta.CgroupDir
+		go func(pod *corev1.Pod, podCgroupDir string) {
+			defer wg.Done()
+			c.collectSinglePodCPI(string(pod.UID))
+			c.collectSinglePodPSI(pod, podCgroupDir)
+		}(pod, podCgroupDir)
+	}
+	wg.Wait()
+	klog.V(5).Infof("collectPodMetrics for time window %s finished at %s, pod num %d",
+		timeWindow, time.Now(), len(podMetas))
+}
+
+func (c *performanceCollector) collectSinglePodPSI(pod *corev1.Pod, podCgroupDir string) {
+	collectTime := time.Now()
+	psiMap, err := util.GetPodPSI(podCgroupDir)
+	if err != nil {
+		klog.Errorf("collect pod %s psi err: %v", pod.Name, err)
+		return
+	}
+	containerPsiMetric := &metriccache.PodInterferenceMetric{
+		MetricName: metriccache.MetricNamePodPSI,
+		PodUID:     string(pod.UID),
+		MetricValue: &metriccache.PSIMetric{
+			SomeCPUAvg10: psiMap["SomeCPUAvg10"],
+			SomeMemAvg10: psiMap["SomeMemAvg10"],
+			SomeIOAvg10:  psiMap["SomeIOAvg10"],
+			FullCPUAvg10: psiMap["FullCPUAvg10"],
+			FullMemAvg10: psiMap["FullMemAvg10"],
+			FullIOAvg10:  psiMap["FullIOAvg10"],
+		},
+	}
+	err = c.metricCache.InsertPodInterferenceMetrics(collectTime, containerPsiMetric)
+	if err != nil {
+		klog.Errorf("insert pod psi metrics failed, err %v", err)
+	}
+	metrics.RecordPodPSI(string(pod.UID), "some", "cpu", psiMap["SomeCPUAvg10"])
+	metrics.RecordPodPSI(string(pod.UID), "some", "mem", psiMap["SomeMemAvg10"])
+	metrics.RecordPodPSI(string(pod.UID), "some", "io", psiMap["SomeIOAvg10"])
+	metrics.RecordPodPSI(string(pod.UID), "full", "cpu", psiMap["FullCPUAvg10"])
+	metrics.RecordPodPSI(string(pod.UID), "full", "mem", psiMap["FullMemAvg10"])
+	metrics.RecordPodPSI(string(pod.UID), "full", "io", psiMap["FullIOAvg10"])
+}
+
+func (c *performanceCollector) collectSinglePodCPI(podUid string) {
```

Comment:
why we need to do this `collectSinglePodCPI`, for prometheus metrics? this can be sum on grafana